docs/articles/documentation/test-api/intercepting-and-mocking-http-requests.md

```diff
@@ -0,0 +1,547 @@
+
+To create a request logger, use the `RequestLogger` constructor.
+
+```text
+RequestLogger([filter] [, options])
+```
+
+Parameter    | Type | Description | Default
+------------ | ---- | ----------- | --------
+`filter`&#160;*(optional)*  | String &#124; RegExp &#124; Object &#124; Predicate | Specifies which requests should be tracked by the logger. See [Specifying Which Requests are Handled by the Hook](#specifying-which-requests-are-handled-by-the-hook). | All requests are tracked
+`options`&#160;*(optional)* | Object | Options that define how the requests and responses are logged. | See below
+
+The `options` parameter contains the following options.
+
+Option | Type | Description   | Default
+------ | ---- | ------------- | ---------
+`logRequestHeaders` | Boolean | Specifies whether the request headers should be logged. | `false`
+`logRequestBody` | Boolean | Specifies whether the request body should be logged. | `false`
+`stringifyRequestBody` | Boolean | Specifies whether the request body should be stored as a string or an object. | `false`
+`logResponseHeaders` | Boolean | Specifies whether the response headers should be logged. | `false`
+`logResponseBody` | Boolean | Specifies whether the response body should be logged. | `false`
+`stringifyResponseBody` | Boolean | Specifies whether the response body should be stored as a string or an object. | `false`
+
+```js
+import { RequestLogger } from 'testcafe';
+
+const simpleLogger = RequestLogger('http://example.com');
+const headerLogger = RequestLogger(/testcafe/, {
+    logRequestHeaders: true,
+    logResponseHeaders: true
+});
+```
+
+To enable the logger to track the requests, [attach it to a test or fixture](#attaching-hooks-to-tests-and-fixtures).
+
+By default, `RequestLogger` stores the following parameters.
+
+* The URL where the request is sent.
+* The request's HTTP method.
+* The status code received in the response.
+* The user agent that sent the request.
+* The ID of the test run that sent the request.
+
+To access data stored by the logger, use its API.
+
+### Methods
+
+Method | Return Type | Description
+------ | ----------- | -------------
+`contains(predicate)` | Promise | Returns whether the logger contains a request that matches the predicate.
+`count(predicate)`    | Promise | Returns the number of requests that match the predicate.
+`clear()`             | None    | Clears all logged requests.
+
+The `predicate` functions take a single parameter - the `Request` object.
+
+### Properties
+
+Property | Type | Description
+-------- | ---- | -----------
+`requests` | Array of Request | Returns an array of logged requests.
+
+### The Request Object
+
+This object represents a request-response pair.
+
+Property | Type | Description
+-------- | ---- | -----------
+`id`         | String | The ID of the request (generated by the logger).
+`testRunId`  | String | The ID of the test run that sent the request. Use it to identify the test in which the request originated.
```

Comment:
I think we should give an example of using `testRunId`

Reply:
I doubt we should expose `testRunId`. Do you know any cases when a user need it? /cc @miherlosev @VasilyStrelyaev

Reply:
@miherlosev once said that grouping requests by the origin test is a use case

Reply:
What's the real world case?

```js
await t
            .expect(logRecord.id).eql('UnqLnm189')
            .expect(logRecord.testRunId).eql('IwQA12J12')
```

It does not make any sense. It will be always failed

Reply:
ehm, I used to have `console.log` here instead of assertions, I guess I will go back to them.

Reply:
`I used to have console.log here instead of assertions`

At first please describe why someone need it.

Reply:
Don't need describe